app/views/articles/_full_comment_area.html.erb

```diff
@@ -55,7 +55,10 @@
 
         <div class="comments" id="comment-trees-container">
           <% if @article.comments_count > 0 %>
-            <%= render partial: "articles/comment_tree", collection: Comment.tree_for(@article, @comments_to_show_count, @comments_order), as: :comment_node, cached: proc { |comment, _sub_comments| [comment, @comments_order] } %>
+            <%= render partial: "articles/comment_tree",
+                       collection: article_comment_tree(@article, @comments_to_show_count, @comments_order),
+                       as: :comment_node,
+                       cached: proc { |comment, _sub_comments| [comment, @comments_order] } %>
```

Comment:
I noticed some inconsistencies with how [this other change](https://github.com/forem/forem/pull/18517) is rolling out in production, and I think it's that this `cached:` condition _also_ has to account for `user_signed_in?` — so might as well deal with that here and will kill two birds with one stone.

https://guides.rubyonrails.org/caching_with_rails.html#collection-caching

Reply:
:+1: